### ISSUE TITLE
Provide API to get collection identity from model instance

### ISSUE THREAD

**Issue by huafu** [balderdashy/waterline#692]
This is more a question than an issue, but if no answer it'll be an issue.

How can I get the model name (identity) or the model pseudo-class object (from `sails.models`) of a given record object?

[SO question](http://stackoverflow.com/questions/26446206/how-to-get-the-model-name-or-model-class-of-a-waterline-record)

**Comment by devinivy** [comment]
It looks like this was answered on Stack Overflow, on the link you provided.  If you think there's a potential feature request here, please start a new issue with some proposed specs!

**Comment by huafu** [comment]
Well the only correct (ie working for my described case) is mine [there](http://stackoverflow.com/a/26447442/1603618), but that is a dirty workaround I believe.

Today there is no way, outside of a controller for a given model, to know what mode instance is a record except that hack I described.

I don't think that it's a good way of doing things, so yeah a PR with a better way would be great, tho I have, today, no time for this. I guess this issue should remain open until me or someone else provide this PR for such a feature.

**Comment by devinivy** [comment]
Given that, I've repurposed this issue as a feature request.  Hope that's cool!

**Comment by huafu** [comment]
Ha, thanks that's perfect!

**Comment by tjwebb** [comment]
You can just require actionUtil and use its methods. 

e.g. https://github.com/tjwebb/sails-permissions/blob/master/api/policies/ModelPolicy.js#L8

**Comment by dmarcelino** [comment]
Another workaround (if you don't have too many models) is to add an instance method that returns the name of the model, for example:
```javascript
// user.js
module.exports = {
 attributes : {
      model : function(){ return 'user'; }
}
```

**Comment by huafu** [comment]
@tjwebb no because that means you need the `request` object and it'll only give you `user` because the request has been done in the `user` controller for example

@dmarcelino that would be a clean solution but it'd be better to have a waterline helper for this instead of having to add a method to all models ;-)

**Comment by dmarcelino** [comment]
> @dmarcelino that would be a clean solution but it'd be better to have a waterline helper for this instead of having to add a method to all models ;-)

I did say it was a workaround :wink: 

It just occurred to me, a simple "solution" to this would be to overwrite the records `toString` to something like:
```
[object modelName]
```

Are we currently doing anything with the `toString` of the records?

**Comment by sailsbot** [comment]
Thanks for posting, @huafu.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
